woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -132,6 +132,105 @@ def test_accessor_schema_property(sample_df):
     assert sample_df.ww._schema == sample_df.ww.schema
 
 
+def test_set_accessor_name(sample_df):
+    df = sample_df.copy()
+    error = re.escape("Woodwork not initialized for this DataFrame. Initialize by calling DataFrame.ww.init")
+    with pytest.raises(WoodworkNotInitError, match=error):
+        df.ww.name
+    with pytest.raises(WoodworkNotInitError, match=error):
+        df.ww.name = 'name'
+
+    df.ww.init()
+    assert df.ww.name is None
+    df.ww.name = 'name'
+    assert df.ww.schema.name == 'name'
+    assert df.ww.name == 'name'
+
+
+def test_rename_init_with_name(sample_df):
+    df = sample_df.copy()
+    df.ww.init(name='name')
+    assert df.ww.name == 'name'
+    df.ww.name = 'new_name'
+    assert df.ww.schema.name == 'new_name'
+    assert df.ww.name == 'new_name'
+
+
+def test_name_error_on_init(sample_df):
+    err_msg = "TableSchema name must be a string"
```

Comment:
It seems to me like this error message isn't generic enough. This could happen when setting a name for either the schema and the accessor (like you are doing here), so it could be confusing to reference `TableSchema`. Maybe we should just say `Name must be a string` instead?